### PR TITLE
emitter: constexpr/misc changes

### PR DESCRIPTION
**Pull request by lioncash**
Derived partially from an existing pull request that's up for Dolphin. Just figured I'd put it up here as well in case it's desirable.

### CODE REVIEW

File: src/common/x64/emitter.h [hunk 1]
```diff
@@ -238,56 +229,60 @@ struct OpArg
...
+constexpr OpArg UImmAuto(u32 imm) {
     return OpArg(imm, imm >= 128 ? SCALE_IMM32 : SCALE_IMM8);
 }
-inline OpArg SImmAuto(s32 imm) {
+constexpr OpArg SImmAuto(s32 imm) {
     return OpArg(imm, (imm >= 128 || imm < -128) ? SCALE_IMM32 : SCALE_IMM8);
 }
 
+template <typename T>
+OpArg ImmPtr(const T* imm)
```
Comment (MerryMage) [inline src/common/x64/emitter.h]: ImmPtr used to be inline, but here it is no longer. Was that intentional?

Comment (lioncash) [inline src/common/x64/emitter.h]: Yes. Inline isn't necessary on function templates that aren't a full specialization.

**Comment by MerryMage** [comment]
LGTM

**Comment by bunnei** [comment]
LGTM

**Pull request merged**